hubstorage/resourcetype.py

```diff
@@ -65,24 +90,24 @@ def _add_resume_param(self, lastline, offset, params):
     def iter_values(self, *args, **kwargs):
         """Reliably iterate through all data as python objects
 
-        calls iter_json, decoding the results
+        calls either iter_json or iter_msgpack, decoding the results
         """
+        if self._supports_msgpack:
+            return mpdecode(self.iter_msgpack(*args, **kwargs))
         return jldecode(self.iter_json(*args, **kwargs))
 
-    def iter_json(self, _path=None, requests_params=None, **apiparams):
-        """Reliably iterate through all data as json strings"""
-        requests_params = dict(requests_params or {})
-        requests_params.setdefault('method', 'GET')
-        requests_params.setdefault('stream', True)
+    def retry(self, iter_callback, resume=False, _path=None, requests_params=None, **apiparams):
+        """Reliable iterate through all data calling iter_callback"""
         lastexc = None
-        line = None
+        chunk = None
         offset = 0
         for attempt in xrange(self.MAX_RETRIES):
-            self._add_resume_param(line, offset, apiparams)
+            if resume:
+                self._add_resume_param(chunk, offset, apiparams)
```

Comment:
it seems to me that you better moving `_add_key_meta` above this condition instead of keeping it inside _add_resume_param and duplicating it in https://github.com/scrapinghub/python-hubstorage/pull/59/files?diff=split#diff-5ced7809b9453346a782751d6b9a698cR134